core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.ts

```diff
@@ -21,23 +26,52 @@ import { WorkflowActionService } from "../../service/workflow-graph/model/workfl
 import { WorkflowUtilService } from "../../service/workflow-graph/util/workflow-util.service";
 import { WorkflowStatusService } from "../../service/workflow-status/workflow-status.service";
 import { ExecutionState, OperatorState } from "../../types/execute-workflow.interface";
-import { OperatorLink, OperatorPredicate, Point } from "../../types/workflow-common.interface";
+import { Breakpoint, OperatorLink, OperatorPredicate, Point } from "../../types/workflow-common.interface";
 import { auditTime, filter, map, takeUntil } from "rxjs/operators";
 import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
 import { UndoRedoService } from "../../service/undo-redo/undo-redo.service";
 import { WorkflowCollabService } from "../../service/workflow-collab/workflow-collab.service";
 import { WorkflowVersionService } from "../../../dashboard/service/workflow-version/workflow-version.service";
 
 // This type represents the copied operator and its information:
+// - operatorID: ID for the operator
 // - operator: the copied operator itself, and its properties, etc.
 // - position: the position of the copied operator on the workflow graph
-// - pastedOperators: a list of operators that are created out of the original operator,
-//   including the operator itself.
+// - layer: layer number of the operator
 type CopiedOperator = {
+  operatorID: string;
   operator: OperatorPredicate;
   position: Point;
   layer: number;
-  pastedOperatorIDs: string[];
+};
+
+type OperatorPositions = {
+  [key: string]: Point;
+};
+
+type BreakpointWithLinkID = {
+  [key: string]: Breakpoint;
+};
+
+type Link = {
```

Comment:
The reason I defined the new type Link is because the OperatorLink type is read only, and I have some codes below that needs to edit the Link type directly, as shown in the screenshot below. That is, for now, the linksCopy[link.linkID]'s type is Link, and if I change it to OperatorLink, I can't change the source or target property.
![image](https://user-images.githubusercontent.com/78711914/187036448-e835f2d1-cdc4-4571-a593-e0ca511f941c.png)

However, if there's a way to work around it, please let me know.